lib/lib/model/season.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:f1_application/lib/model/driver.dart';
+
+class Season {
+  Season({
+    required this.year,
+    required this.wikiURL,
+    required this.driverIds,
+  });
+
+  String year;
+  String wikiURL;
+  List<String> driverIds;
+
+  static Season fromJson(Map<String, dynamic> json) => Season(
+    year: json[r'season'],
+    wikiURL: json[r'url'],
+    driverIds: json[r'driverIds'] ?? []
+  );
+
+  static List<Season> listFromJson(List<dynamic> json) =>
+      json.map((e) => Season.fromJson(e)).toList(growable: true);
+
+  String get shortYear => "'${year.toString().substring(2, 4)}";
+
+  void addDrivers(List<Driver> drivers) {
+    for (Driver driver in drivers) {
+      addDriver(driver);
+    }
+  }
+
+  void addDriver(Driver driver) {
+    if (!driverIds.contains(driver.id)) driverIds.add(driver.id);
+  }

```

Comment:
Ez a két függvény és a `driverIds` lista valójában nincs használva semmire és ha model osztályokról beszélünk, talán nem is célszerű ezt a két dolgot összemosni. Szóban szívesen elmondom miért gondolom így, mert egyszerűbb. Addig nyugodtan hagyd nyitva ezt a kommentet de a két kijelölt függvényt és a `driverIds` propertyt, plusz mindent ami  ezt hivatkozza, nyugodtan törölj ki, nem fog elrontani semmit